alica_engine/include/engine/blackboard/Blackboard.h

```diff
@@ -136,8 +136,11 @@ class BlackboardImpl
     {
         using Type = std::decay_t<T>;
         // Promote signed integers to int64 & unsigned integers to uint64 to avoid compiler error in variant constructor/assignment operator (due to ambiguity)
-        using PromotedType = std::conditional_t<std::is_integral_v<Type> && !std::is_same_v<Type, bool>,
+        using IntPromotedType = std::conditional_t<std::is_integral_v<Type> && !std::is_same_v<Type, bool>,
                 std::conditional_t<std::is_signed_v<Type>, int64_t, uint64_t>, Type>;
+        // Promote float to double. Note: we promote to double & not long double for backwards compatibility (otherwise variant type would need to be changed
+        // resulting in long double being treated as a known type instead of unknonwn (std::any) & all get<double> statements changing to get<long double>)
```

Comment:
```suggestion
        // resulting in long double being treated as a known type instead of unknown (std::any) & all get<double> statements changing to get<long double>)
```